Jenkinsfile

```diff
@@ -1,47 +1,94 @@
+// Define variables
+HOME = "/home/saluser"
+
 pipeline{
-    agent{
+    parameters {
+        string defaultValue: 'develop', description: 'The version of the Docker image to use for the build.', name: 'image_version', trim: true
+    }
+    agent {
         docker {
-            alwaysPull true
-            image 'lsstts/develop-env:develop'
-            args "-u root --entrypoint=''"
-        }
+            image 'ts-dockerhub.lsst.org/salobj:$image_version'
+            args '-u root --entrypoint="" ' +
+            '-e LSST_DDS_PARTITION_PREFIX=citest -v ${WORKSPACE}:' + HOME + '/repos/ts_SalMultiLanguageTests '
+            label 'Node1_4CPU || Node3_4CPU'
+         }
     }
     environment {
         user_ci = credentials('lsst-io')
         LTD_USERNAME="${user_ci_USR}"
         LTD_PASSWORD="${user_ci_PSW}"
     }
     stages{
+        stage("Build Test CSC libraries") {
+            steps {
+                 withEnv(["HOME=/home/saluser"]) {
+                    sh """
+                    echo 'Setup the environment'
+                    set +x
+                    source $HOME/.setup.sh
+                    set -x
+                    echo 'Update ts_idl/develop'
+                    cd $HOME/repos/ts_idl
+                    /home/saluser/.checkout_repo.sh develop
+                    git pull
+                    echo 'Update ts_xml/develop'
+                    cd $HOME/repos/ts_xml
+                    /home/saluser/.checkout_repo.sh develop
+                    git pull
+                    echo 'Update ts_sal/develop'
+                    cd $HOME/repos/ts_sal
+                    /home/saluser/.checkout_repo.sh develop
+                    git pull
+                    cd test
+                    echo 'Copy XML files'
+                    cp -r $HOME/repos/ts_xml/sal_interfaces/Test/Test*.xml $HOME/repos/ts_sal/test
+                    cp -r $HOME/repos/ts_xml/sal_interfaces/SALSubsystems.xml $HOME/repos/ts_sal/test
+                    cp -r $HOME/repos/ts_xml/sal_interfaces/SALGenerics.xml $HOME/repos/ts_sal/test
+                    cp -r $HOME/repos/ts_xml/VERSION $HOME/repos/ts_sal/test
+                    echo 'Validate'
+                    salgenerator Test validate
+                    echo 'IDL'
+                    salgenerator Test sal idl
+                    echo 'C++'
+                    salgenerator Test sal cpp
+                    echo 'Java'
+                    salgenerator Test sal java version=pre${BUILD_NUMBER}-SNAPSHOT
+                    echo 'Maven'
+                    salgenerator Test maven version=pre${BUILD_NUMBER}-SNAPSHOT
+                    echo 'Lib'
+                    salgenerator Test lib
+                    echo 'Copy controllers'
+                    cp $HOME/repos/ts_sal/bin/minimal_*_controller.sh $HOME/repos/ts_SalMultiLanguageTests/tests/controllers
+                    """
```

Comment:
I still *strongly* suggest adding `make_idl_files.py --keep Test`.

If you don't do that then you have to do all this:
* Generate the revcoded Test file (as you are already doing)
* Move or copy it to the `idl` directory in the `ts_idl` package, which I don't see you doing.

If you miss the second step then then you risk the salobj code using a version of the Test topics that is incompatible with C++ and Java. Using `make_idl_files.py` is the standard way to handle this, and it is known to work well.

Reply:
Done.